### ISSUE TITLE
Global `escape` method is deprecated

### ISSUE THREAD

**Issue by leonderijke** [auth0/jwt-decode#7]
According to [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/escape) the global `escape` method used in `base64_url_decode.js` is deprecated and should not be relied on.

**Comment by jfromaniello** [comment]
That's right, do you know any other alternative (maybe on npm) to this exact functionality?

thanks

**Comment by leonderijke** [comment]
No, I don't. MDN says we should use `encodeURI` or `encodeURIComponent` instead, but I'm not sure if the functionality is the same.

**Comment by jfromaniello** [comment]
Yes, I was looking at it and I copied the code I used from the same mdn :)

https://developer.mozilla.org/es/docs/Web/API/WindowBase64/btoa#Unicode_Strings

basically we need a way to base64 decode unicoded strings in the browser without using `escape`.

**Comment by wjbryant** [comment]
If you don't want to use `escape` and `unescape`, you can use the following functions instead which are based on [code from MDN](https://developer.mozilla.org/en-US/docs/Web/API/WindowBase64/Base64_encoding_and_decoding) and [Unibabel.js](https://github.com/coolaj86/unibabel-js).

```javascript
// based on code from Unibabel
/*! Unibabel | https://github.com/coolaj86/unibabel-js
(c) 2015 AJ ONeal | http://opensource.org/licenses/mit */

function b64EncodeUnicode(str) {
    return btoa(encodeURIComponent(str).replace(/%([0-9A-F]{2})/g, function(match, p1) {
        return String.fromCharCode(parseInt(p1, 16));
    }));
}

function b64DecodeUnicode(str) {
    return decodeURIComponent(atob(str).replace(/(.)/g, function (m, p) {
        var code = p.charCodeAt(0).toString(16).toUpperCase();
        if (code.length < 2) {
            code = '0' + code;
        }
        return '%' + code;
    }));
}
```

**Comment by jfromaniello** [comment]
@wjbryant thanks a lot, `b64DecodeUnicode` was exactly what I was looking for.

I have published as `v1.5.0`
